File: decompiled/4.0.2/be/ehealth/technicalconnector/validator/impl/XMLValidatorImpl.java
package be.ehealth.technicalconnector.validator.impl;

import be.ehealth.technicalconnector.exception.TechnicalConnectorException;
import be.ehealth.technicalconnector.exception.TechnicalConnectorExceptionValues;
import be.ehealth.technicalconnector.validator.ValidatorHelper;
import be.ehealth.technicalconnector.validator.XMLValidator;
import java.util.HashMap;
import java.util.Map;

public class XMLValidatorImpl implements XMLValidator {
   protected static final Map<Class, String> XSD_FILE_LOCATION_FOR_CLASS_MAP = new HashMap();

   public XMLValidatorImpl() {
   }

   public void validate(Object xmlObject) throws TechnicalConnectorException {
      if (xmlObject != null) {
         ValidatorHelper.validate(xmlObject, xmlObject.getClass(), this.getXsdFileLocationForXmlObject(xmlObject));
      } else {
         throw new TechnicalConnectorException(TechnicalConnectorExceptionValues.ERROR_XML_INVALID, new Object[]{"xml object had null value"});
      }
   }

   private String getXsdFileLocationForXmlObject(Object xmlObject) throws TechnicalConnectorException {
      if (xmlObject != null && XSD_FILE_LOCATION_FOR_CLASS_MAP.containsKey(xmlObject.getClass())) {
         return (String)XSD_FILE_LOCATION_FOR_CLASS_MAP.get(xmlObject.getClass());
      } else {
         throw new TechnicalConnectorException(TechnicalConnectorExceptionValues.ERROR_XML_INVALID, new Object[]{"no xsd source defined for xmlObject " + xmlObject});
      }
   }
}

File: src/main/kotlin/org/taktik/connector/technical/validator/impl/XMLValidatorImpl.kt
package org.taktik.connector.technical.validator.impl

import org.apache.commons.lang.mutable.Mutable
import org.taktik.connector.technical.exception.TechnicalConnectorException
import org.taktik.connector.technical.exception.TechnicalConnectorExceptionValues
import org.taktik.connector.technical.validator.ValidatorHelper
import org.taktik.connector.technical.validator.XMLValidator
import java.util.HashMap

open class XMLValidatorImpl : XMLValidator {

    @Throws(TechnicalConnectorException::class)
    override fun validate(xmlObject: Any?) {
        if (xmlObject != null) {
            ValidatorHelper.validate(xmlObject, xmlObject.javaClass, this.getXsdFileLocationForXmlObject(xmlObject))
        } else {
            throw TechnicalConnectorException(TechnicalConnectorExceptionValues.ERROR_XML_INVALID, "xml object had null value")
        }
    }

    @Throws(TechnicalConnectorException::class)
    private fun getXsdFileLocationForXmlObject(xmlObject: Any?): String {
        return if (xmlObject != null && XSD_FILE_LOCATION_FOR_CLASS_MAP.containsKey(xmlObject.javaClass)) {
            XSD_FILE_LOCATION_FOR_CLASS_MAP[xmlObject.javaClass] as String
        } else {
            throw TechnicalConnectorException(TechnicalConnectorExceptionValues.ERROR_XML_INVALID, "no xsd source defined for xmlObject " + xmlObject!!)
        }
    }

    companion object {
        val XSD_FILE_LOCATION_FOR_CLASS_MAP: MutableMap<Class<*>, String> = HashMap()
    }
}
